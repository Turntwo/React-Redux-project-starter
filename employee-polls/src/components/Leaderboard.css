.leader-table {
  width: 75%;
  margin: auto;
}

.leader-table-header {
  border: 1px solid;
  background-color: lightcyan;
}

.leaderboard-image {
  height: 24px;
}
