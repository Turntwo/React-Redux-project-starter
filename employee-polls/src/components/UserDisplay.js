const UserDisplay = ({ user }) => {
  return (
    <tr>
      <td>
        <img className="leaderboard-image" src={user.avatarURL} alt=""></img>
        {user.name}
      </td>
      <td data-testid="answers">{Object.keys(user.answers).length}</td>
      <td data-testid="questions">{user.questions.length}</td>
    </tr>
  );
};

export default UserDisplay;
