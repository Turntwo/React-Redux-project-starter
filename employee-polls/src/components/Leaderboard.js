import "./Leaderboard.css";
import { connect } from "react-redux";
import UserDisplay from "./UserDisplay";

const Leaderboard = ({ users }) => {
  const getCounts = (user) => {
    return Object.keys(user.answers).length + user.questions.length;
  };

  return (
    <>
      <h1>Leaderboard</h1>
      <table className="leader-table">
        <thead>
          <tr>
            <th className="leader-table-header">Users</th>
            <th className="leader-table-header">Answered</th>
            <th className="leader-table-header">Created</th>
          </tr>
        </thead>
        <tbody>
          {Object.values(users)
            .sort((a, b) => getCounts(b) - getCounts(a))
            .map((user) => (
              <UserDisplay key={user.id} user={user} />
            ))}
        </tbody>
      </table>
    </>
  );
};

const mapStateToProps = (state, ownProps) => {
  return {
    users: state.users,
    loggedInUser: state.loggedInUser,
  };
};

export default connect(mapStateToProps)(Leaderboard);
