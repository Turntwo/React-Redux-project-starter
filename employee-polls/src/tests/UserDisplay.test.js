import * as React from "react";
import { render } from "@testing-library/react";
import UserDisplay from "../components/UserDisplay";
import Leaderboard from "../components/Leaderboard";
import { createStore } from "redux";
import reducer from "../reducers/index";
import middleware from "../middleware/index";
import { Provider } from "react-redux";

describe("UserDisplay", () => {
  const user = {
    id: "sarahedo",
    password: "password123",
    name: "Sarah Edo",
    avatarURL: "http://avatar.windsor.io/SarahEdo",
    answers: {
      "8xf0y6ziyjabvozdd253nd": "optionOne",
      "6ni6ok3ym7mf1p33lnez": "optionOne",
      am8ehyc8byjqgar0jgpub9: "optionTwo",
      loxhs1bqm25b708cmbf3g: "optionTwo",
    },
    questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
  };

  it("will match snapshot with no name", async () => {
    const view = render(<UserDisplay key={user.id} user={user} />);
    expect(view).toMatchSnapshot();
  });

  it("will display count columns", async () => {
    const view = render(<UserDisplay key={user.id} user={user} />);
    expect(view.getByTestId("questions")).toBeInTheDocument();
    expect(view.getByTestId("answers")).toBeInTheDocument();
  });
});
