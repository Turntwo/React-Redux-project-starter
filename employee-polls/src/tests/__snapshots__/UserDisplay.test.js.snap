// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDisplay will match snapshot with no name 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr>
        <td>
          <img
            alt=""
            class="leaderboard-image"
            src="http://avatar.windsor.io/SarahEdo"
          />
          Sarah Edo
        </td>
        <td
          data-testid="answers"
        >
          4
        </td>
        <td
          data-testid="questions"
        >
          2
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr>
      <td>
        <img
          alt=""
          class="leaderboard-image"
          src="http://avatar.windsor.io/SarahEdo"
        />
        Sarah Edo
      </td>
      <td
        data-testid="answers"
      >
        4
      </td>
      <td
        data-testid="questions"
      >
        2
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
